flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/AbstractFileStoreWrite.java

```diff
@@ -104,21 +108,30 @@ public void compact(BinaryRowData partition, int bucket) throws Exception {
     }
 
     @Override
-    public List<FileCommittable> prepareCommit(boolean endOfInput) throws Exception {
+    public List<FileCommittable> prepareCommit(boolean endOfInput, long commitIdentifier)
+            throws Exception {
+        long latestCommittedIdentifier =
+                snapshotManager
+                        .latestSnapshotOfUser(commitUser)
```

Comment:
If it is the first commit, all snapshots will be scanned? Can we avoid this behavior?